src/State/Scope.php

```diff
@@ -96,6 +102,38 @@ public function getTags(): array
         return $this->tags->toArray();
     }
 
+    /**
+     * Sets context data with the given name.
+     *
+     * @param string $name  The name that uniquely identifies the context
+     * @param array  $value The value
+     *
+     * @return self
+     */
+    public function setContext(string $name, array $value): self
+    {
+        if (!isset($this->contexts[$name])) {
+            $this->contexts[$name] = new \stdClass();
+        }
+        foreach ($value as $arrayKey => $arrayValue) {
+            $this->contexts[$name]->{$arrayKey} = $arrayValue;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Gets the tags contained in the contexts context.
+     *
+     * @return array<string, array>
```

Comment:
The typehint doesn't looks correct to me, I would expect something like `array<string, \stdClass>`

Reply:
Correct, done